sqlmesh/core/model/decorator.py

```diff
@@ -27,6 +31,18 @@ def __init__(self, name: str, is_sql: bool = False, **kwargs: t.Any) -> None:
         if not is_sql and "columns" not in kwargs:
             raise ConfigError("Python model must define column schema.")
 
+        kind = kwargs.get("kind", None)
+        if kind is not None:
+            if isinstance(kind, _ModelKind):
+                logger.warning(
+                    f"""Python model "{name}"'s `kind` argument was passed a SQLMesh `{type(kind).__name__}` object. This is no longer supported - provide a dictionary instead."""
+                )
```

Comment:
The `_ModelKind` object will still work until we actually deprecate this, right? Should we instead say that we plan to deprecate it instead of "is no longer supported"?

Reply:
Good call. @tobymao at some point are we going to error if we get a model instance or will we always allow with warning?

If we'll always allow it, maybe better wording is "may exhibit unexpected behavior"?

Reply:
yea the latter, we encourage people to not use the python api and it's not really supported